### PR TITLE
DEPR: remove tm.makePanel and all usages

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/frame/test_reshape.py [hunk 1]
```diff
@@ -49,14 +48,6 @@ def test_pivot(self):
         assert pivoted.index.name == 'index'
         assert pivoted.columns.names == (None, 'columns')
 
-        with catch_warnings(record=True):
-            # pivot multiple columns
-            simplefilter("ignore", FutureWarning)
-            wp = tm.makePanel()
-            lp = wp.to_frame()
-            df = lp.reset_index()
-            tm.assert_frame_equal(df.pivot('major', 'minor'), lp.unstack())
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_reshape.py]: would this test not need to replaced with a similar test not dependent on tm.makePanel?

**Review by WillAyd — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
@WillAyd I get a `unexpectedly closed the connection` from codecov.io so i cant see the detailed report, but i'm not sure why you would approve this with a drop in code coverage in core modules other than panel. Am i missing something?

**Comment by WillAyd** [comment]
Didn’t look at the coverage report just code. Seemed to only pull out deprecated parts. 

Thanks for questioning though - is your concern that we are dropping tests for non-deprecated parts of code base?

**Comment by simonjayhawkins** [comment]
my concern is that makePanel.to_fame is part of a least one test setup and the tested behaviour maybe non-panel related.

**Comment by jbrockmendel** [comment]
> my concern is that makePanel.to_fame is part of a least one test setup and the tested behaviour maybe non-panel related.

I'm guessing you're referring to the test in test_reshape.  There is another test where I manually reconstructed the equivalent of `tm.makePanel().to_frame()`.  We could put that code into `tm` and re-use it in the reshape test.  Would that handle your concern?

**Comment by simonjayhawkins** [comment]
considering how difficult it can sometimes be to get a single line of code into the codebase, there is an incredible amount of value in the 1600+ lines of code being removed. being in the dark about the code coverage at the moment does make me concerned about a potential loss of value.

**Comment by simonjayhawkins** [comment]
> I'm guessing you're referring to the test in test_reshape. There is another test where I manually reconstructed the equivalent of `tm.makePanel().to_frame()`. We could put that code into `tm` and re-use it in the reshape test. Would that handle your concern?

@jbrockmendel: probably not necessary, i briefly had access to the coverage report and I couldn't see that as being an issue.

@WillAyd: lgtm

**Comment by jreback** [comment]
@simonjayhawkins we have *wanted* to remove this code for quite some time :-D

and removing codes IS much easier to approve than adding, reducing complexity is always a +1 from me (granted *removing* tests requires a fair amount of scrutiny).

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**